### PR TITLE
Add Stories RSS feed link to HTML head

### LINKED ISSUE #3136: Add Stories RSS Feed to HTML head
## Feature Description

In #3086 there was a question whether there is an RSS feed or not. To make the feed and the stories content in general easier to discover, let's print the RSS Feed to the HTML `<head>`.

This is essentially replicating https://github.com/ampproject/amp-wp/commit/6ec66a49e6ebebae28b1afa870fdc0866c20b1d6

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* Adapt https://github.com/ampproject/amp-wp/commit/6ec66a49e6ebebae28b1afa870fdc0866c20b1d6

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Add feed link in the header. 
Also format the RSS feed output as an image. 

## Relevant Technical Choices

To repeated code, I moved the logic to render in feed for stories to a trait, so we can reuse. It isn't the cleanest thing, but it works. 

I looked at enabling feeds with adding

```
'rewrite' => [
    'feeds' => true,
]
```

But this didn't work without enabling archives. 

I also moved the logic for render the story in feed, into its own file.  

## Testing Instructions

---

Fixes #3136

### CODE REVIEW

File: includes/Discovery.php [hunk 1]
```diff
@@ -219,4 +221,45 @@ public function print_twitter_metadata() {
...
+		}
+
+		$feed_url = add_query_arg(
+			'post_type',
+			Story_Post_Type::POST_TYPE_SLUG,
+			get_feed_link()
+		);
+
+		$name = '';
+		if ( property_exists( $post_type_object->labels, 'name' ) ) {
```
Comment (swissspidy) [inline includes/Discovery.php:247]: I don't think this is ever `false`, so seems unnecessary.

Comment (spacedmonkey) [reply includes/Discovery.php:247]: PHPstan flagged this one 🤷‍♂️

**Review by swissspidy — CHANGES_REQUESTED** [review]
As discussed, we need a more holistic approach for the whole rendering, for which we need a new issue.

Could you update this PR so that it only includes the `print_feed_link` addition?

Everything else should be a separate PR in the future, once we've settled on an approach for the rendering.

**Comment by spacedmonkey** [comment]
@swissspidy Can you re-review.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**